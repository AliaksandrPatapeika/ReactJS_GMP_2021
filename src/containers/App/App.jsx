import './App.less';

import React from 'react';
import {Provider} from 'react-redux';
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom';

import store from '../../store';
import ErrorBoundary from '../ErrorBoundary';
import HomePage from '../HomePage';
import MovieDetailsPage from '../MovieDetailsPage';
import PageNotFound from '../PageNotFound';
import SearchPage from '../SearchPage';

const App = () => (
  <Provider store={store}>
    <ErrorBoundary>
      <Router>
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route exact path="/search" component={SearchPage} />
          <Route exact path="/film/:id" component={MovieDetailsPage} />
          <Route path="*" component={PageNotFound} />
        </Switch>
      </Router>
    </ErrorBoundary>
  </Provider>
);

export default App;
