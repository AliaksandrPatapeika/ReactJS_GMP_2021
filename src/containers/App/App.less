@import "../../less/reset";
@import "../../less/variables";
@import "../../less/common";

body {
	background-color: @gray;

	/* Sticky Footer with Flexbox */
	#app {
		display: flex;
		flex-direction: column;
		height: 100%;

		.mainContainer,
		.modalWindowContainer {
			flex: 1 0 auto;
		}

		.footerContainer {
			flex-shrink: 0;
		}
	}
}