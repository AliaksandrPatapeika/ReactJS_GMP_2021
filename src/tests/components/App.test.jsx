import React from 'react';
import renderer from 'react-test-renderer';

import App from '../../components/App';

describe('App', () => {
  test('renders correctly', () => {
    const app = renderer
      .create(<App />)
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
