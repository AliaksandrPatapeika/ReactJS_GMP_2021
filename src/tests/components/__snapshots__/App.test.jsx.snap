// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="test_container"
>
  <h1
    className="test_fonts"
  >
    Hello React !
  </h1>
  <img
    alt="logo"
    src="test-file-stub"
  />
</div>
`;
